modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java

```diff
@@ -1037,8 +1119,105 @@ public void testObjectFieldSettings() throws Exception {
+				"a" + RandomTestUtil.randomString()
+			).objectDefinitionId(
+				objectDefinition.getObjectDefinitionId()
+			).objectFieldSettings(
+				Arrays.asList(
+					_createObjectFieldSetting(
+						ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
+						_listTypeEntryKey),
+					_createObjectFieldSetting(
+						ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+						ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE))
+			).build());
+
+		_assertObjectFieldSetting(
+			ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
+			objectField.getObjectFieldId(), _listTypeEntryKey);
+		_assertObjectFieldSetting(
+			ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+			objectField.getObjectFieldId(),
+			ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE);
+
+		_objectDefinitionLocalService.publishCustomObjectDefinition(
+			TestPropsValues.getUserId(),
 			objectDefinition.getObjectDefinitionId());
+
+		_assertObjectEntryDefaultValue(
+			_listTypeEntryKey, objectField, new HashMap<>());
+
+		objectField.setObjectFieldSettings(
+			Arrays.asList(
+				_createObjectFieldSetting(
+					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE, "text"),
+				_createObjectFieldSetting(
+					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+					ObjectFieldSettingConstants.VALUE_EXPRESSION_BUILDER)));
+
+		objectField = _updateCustomObjectField(objectField);
+
+		_assertObjectFieldSetting(
+			ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
+			objectField.getObjectFieldId(), "text");
+		_assertObjectFieldSetting(
+			ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+			objectField.getObjectFieldId(),
+			ObjectFieldSettingConstants.VALUE_EXPRESSION_BUILDER);
+
+		_assertObjectEntryDefaultValue(
+			_listTypeEntryKey, objectField,
+			HashMapBuilder.<String, Serializable>put(
+				"text", _listTypeEntryKey
+			).build());
+
+		objectField = _addCustomObjectField(
+			new PicklistObjectFieldBuilder(
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).listTypeDefinitionId(
+				_listTypeDefinition.getListTypeDefinitionId()
+			).name(
+				"a" + RandomTestUtil.randomString()
+			).objectDefinitionId(
+				objectDefinition.getObjectDefinitionId()
+			).objectFieldSettings(
+				Arrays.asList(
+					_createObjectFieldSetting(
+						ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
+						_listTypeEntryKey),
+					_createObjectFieldSetting(
+						ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+						ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE))
+			).required(
+				true
+			).state(
+				true
+			).build());
+
+		_assertObjectFieldSetting(
+			ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
+			objectField.getObjectFieldId(), _listTypeEntryKey);
+		_assertObjectFieldSetting(
+			ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
+			objectField.getObjectFieldId(),
+			ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE);
+
+		_assertObjectEntryDefaultValue(
+			_listTypeEntryKey, objectField, new HashMap<>());
```

Comment:
@devpedrotavares this code looks duplicated.. Could you reuse code by creating a private method which will receive the  `required` and `state` parameters? Something like:

```
private ObjectField _addPicklistObjectField(
	ObjectDefinition objectDefinition, boolean required, boolean state) {

	ObjectField objectField = _addCustomObjectField(
		new PicklistObjectFieldBuilder(
		).labelMap(
			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
		).listTypeDefinitionId(
			_listTypeDefinition.getListTypeDefinitionId()
		).name(
			"a" + RandomTestUtil.randomString()
		).objectDefinitionId(
			objectDefinition.getObjectDefinitionId()
		).objectFieldSettings(
			Arrays.asList(
				_createObjectFieldSetting(
					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
					_listTypeEntryKey),
				_createObjectFieldSetting(
					ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
					ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE))
		).required(
			required
		).state(
			state
		).build());

	_assertObjectFieldSetting(
		ObjectFieldSettingConstants.NAME_DEFAULT_VALUE,
		objectField.getObjectFieldId(), _listTypeEntryKey);
	_assertObjectFieldSetting(
		ObjectFieldSettingConstants.NAME_DEFAULT_VALUE_TYPE,
		objectField.getObjectFieldId(),
		ObjectFieldSettingConstants.VALUE_INPUT_AS_VALUE);
	
	return objectField;
}
```